app/models/project.rb

```diff
@@ -137,22 +137,22 @@ def to_hash(recurse = true)
       path: UrlGenerator.new.project_path(self),
       hidden: hidden,
       featured: featured,
-      likeCount: likes.count,
+      likeCount: likes.length,
       content: content,
       timeAgoInWords: time_ago_in_words(created_at),
       createdAt: created_at.strftime('%B %d, %Y'),
       ownerName: owner.nil? ? nil : owner.name,
       ownerUrl: owner.nil? ? nil : UrlGenerator.new.user_url(owner),
-      dataSetCount: data_sets.count,
-      dataSetIDs: data_sets.pluck(:id),
-      fieldCount: fields.count,
+      dataSetCount: data_sets.length,
+      dataSetIDs: data_sets.to_a.map(&:id),
```

Comment:
I disagree :) on a large project with a lot of media sets, eager loading all the information up front would almost definitely be faster than loading piecemeal on each recursive call. Note we're not just talking about 2 fast queries here, altogether without eager loading it is polynomial number of queries (like i mentioned on the issue thread). I'm sacrificing memory here for time, but time is a far more valuable resource than memory. Once everything is loaded into memory, there will be no further object loading downstream.